### Visualization of the datasets

The first step that is generally usefull to do before any other kind of peration is usually to load and visualiza the datasets as they are, this operation can help to discover some kind of pattern that may be useful to analyze later on in the general workflow of the project.

Possibili tematiche da analizzare:
- Correlazione tra vari aspetti dell’uguaglianza di genere: Possiamo esaminare le correlazioni tra vari aspetti dell’uguaglianza di genere. Ad esempio, c’è una correlazione tra la discriminazione attitudinale e i guadagni delle donne? O tra i diritti di proprietà delle donne e la loro rappresentanza nel governo locale? Questo potrebbe richiedere qualche un’analisi statistica approfondita e potrebbe rivelare cose interessanti.

- Analisi temporale: Essndo i nostri dati raccolti spesso nel corso del tempo, potremmo esaminare come questi vari aspetti dell’uguaglianza di genere sono cambiati nel tempo. Ci sono stati progressi in alcune aree ma non in altre?

- Confronto tra paesi: Possiamo confrontare questi vari aspetti dell’uguaglianza di genere tra diversi paesi. Ci sono paesi che si comportano particolarmente bene o male in certi aspetti? Questo potrebbe anche darci l’opportunità di esplorare perché alcuni paesi potrebbero avere migliori risultati in termini di uguaglianza di genere (inferenze sulla sola base di dati statistici a meno che non siano state fatte direttamente negli articoli da cui abbiamo preso i dati le lascerie perdere).

- Modellazione predittiva: Possiamo utilizzare tecniche di apprendimento automatico per creare un modello predittivo. Ad esempio, possiamo cercare di prevedere i guadagni delle donne in base a vari fattori, come la discriminazione attitudinale, i diritti di proprietà, la rappresentanza nel governo locale, ecc. (progettino di machine learning, possiamo rimanere sul semplice ma sarebbe figo)

- Analisi dell’impatto delle politiche: Se disponiamo di dati sulle politiche o sulle leggi relative all’uguaglianza di genere, possiamo esaminare l’impatto di queste politiche sui vari aspetti dell’uguaglianza di genere. (integrazione dei dati con quelli relativi alle legislazioni, potrebbe essere un lavoraccio in quanto dovremmo raccogliere legislazioni di diverse località nel corso del tempo a meno che non vogliamo rimanere su un contesto più ampio e generico, del resto i nostri dati provengono dalle nazioni unite)


In [4]:
import pandas as pd

#Attitudinal discrimination against women engaging in paid work outside the home by sex: 2016 (Percentage)
df1_discrimination = pd.read_csv("../data/Discriminatory_attitudes_towards_women_in_paid_employment/download.csv")
df2_discrimination = pd.read_csv("../data/Discriminatory_attitudes_towards_women_in_paid_employment/download (1).csv").fillna("_")

# Distribution of married women and men aged 15—49 by type of earnings from labour in the last 12 months, developing regions: 2013—2018 (latest available) (Percentage)
#df_earnings mostra un difetto nella sua composizione in quanto sotto la colonna sex in realtà mostra la sdg region
df1_earnings = pd.read_csv("../data/Earnings_of_married_women_and_men_in_employment_by_type_of_remuneration/download.csv")
# Gender gap in any cash and cash-only labor incomes between married women and men aged 15—49 in selected countries in developing regions: 2013—2018 (latest available) (Percentage)
df2_earnings = pd.read_csv("../data/Earnings_of_married_women_and_men_in_employment_by_type_of_remuneration/download (1).csv")

df1_equal_rights = pd.read_csv("../data/equal_rights_of_women_to_the_ownership_and_control_of_land_fao/download.csv")
df2_equal_rights = pd.read_csv("../data/equal_rights_of_women_to_the_ownership_and_control_of_land_fao/download (1).csv")
df3_equal_rights = pd.read_csv("../data/equal_rights_of_women_to_the_ownership_and_control_of_land_fao/download (2).csv")
df4_equal_rights = pd.read_csv("../data/equal_rights_of_women_to_the_ownership_and_control_of_land_fao/download (3).csv")

df1_females_stem = pd.read_csv("../data/female_graduates_in_science_technology_engineering_and_mathematics_at_the_tertiary_level_of_education/download.csv")
df2_females_stem = pd.read_csv("../data/female_graduates_in_science_technology_engineering_and_mathematics_at_the_tertiary_level_of_education/download (1).csv")

df1_unpaid_work = pd.read_csv("../data/time_spent_in_unpaid_work_total_work_burden_and_work_life_balance/download.csv")
df2_unpaid_work = pd.read_csv("../data/time_spent_in_unpaid_work_total_work_burden_and_work_life_balance/download (1).csv")
df3_unpaid_work = pd.read_csv("../data/time_spent_in_unpaid_work_total_work_burden_and_work_life_balance/download (2).csv")

df1_women_decision_makers = pd.read_csv("../data/women_as_decision_makers_in_the_corporate_world/download.csv")
df2_women_decision_makers = pd.read_csv("../data/women_as_decision_makers_in_the_corporate_world/download (1).csv")
df3_women_decision_makers = pd.read_csv("../data/women_as_decision_makers_in_the_corporate_world/download (2).csv")
df4_women_decision_makers = pd.read_csv("../data/women_as_decision_makers_in_the_corporate_world/download (3).csv")
df5_women_decision_makers = pd.read_csv("../data/women_as_decision_makers_in_the_corporate_world/download (4).csv")
df6_women_decision_makers = pd.read_csv("../data/women_as_decision_makers_in_the_corporate_world/download (5).csv")

df1_un_women = pd.read_csv("../data/women_in_local_government_un_women/download.csv")
df2_un_women = pd.read_csv("../data/women_in_local_government_un_women/download (1).csv")

df1_women_police = pd.read_csv("../data/women_in_the_police/download.csv")
df2_women_police = pd.read_csv("../data/women_in_the_police/download (1).csv")

### Analisi di correlazione

In [5]:
mean_per_sdg_region = df1_discrimination.groupby('SDG region')[['% Men', '% Women']].mean().reset_index()
# print(mean_per_sdg_region)
#questa operazione banalmente serve ad ottenere il df2_discrimination che diventa quindi superfluo in quanto mostra la media per ogni regione del pianeta di discriminazone verso la donna nei posti di lavoro



df2_earnings.rename(columns={'Sex': 'SDG region'}, inplace=True)
mean_per_sdg_region2 = df2_earnings.groupby('SDG region')[['Any cash labor income', 'Cash only']].mean().reset_index()
# print(mean_per_sdg_region2)

df1 = pd.DataFrame(df1_discrimination)
df2 = pd.DataFrame(df2_earnings)

df = pd.merge(df1, df2, on='SDG region')

correlation = df.corr()

correlation

C:\Users\mguen\AppData\Local\Temp\ipykernel_17848\3218081672.py:16: FutureWarning: The default value of numeric_only in DataFrame.corr is deprecated. In a future version, it will default to False. Select only valid columns or specify the value of numeric_only to silence this warning.
  correlation = df.corr()


% Men   % Women  Any cash labor income  Cash only
% Men                  1.000000  0.834070               0.125778   0.086807
% Women                0.834070  1.000000               0.111501   0.068956
Any cash labor income  0.125778  0.111501               1.000000   0.863380
Cash only              0.086807  0.068956               0.863380   1.000000

From the correlation matrix, we can conclude that:
- There is a strong positive correlation (0.834070) between the percentage of men and women with discriminatory attitudes towards women. This suggests that in countries where a higher percentage of men have discriminatory attitudes towards women, there is also a higher percentage of women with discriminatory attitudes towards other women. 
- The correlations between discriminatory attitudes and the gender gap in earnings are relatively low (between 0.068956 and 0.125778), suggesting that there is not a strong linear relationship between these two aspects. This does not exclude the possibility of a non-linear relationship or the influence of other variables not included in this analysis.



(da confermare e rivedere le seguenti interpretazioni:
- % Men e % Women hanno una correlazione di 0.834070, il che indica una forte correlazione positiva. Questo suggerisce che nei paesi dove gli uomini mostrano comportamenti discriminatori verso le donne, anche le donne tendono a mostrare comportamenti discriminatori verso altre donne, e viceversa.
- % Men e Any cash labor income hanno una correlazione di 0.125778, che è una correlazione positiva ma debole. Questo indica che non c’è una relazione forte tra gli uomini con comportamenti discriminatori e il divario salariale di genere.
- % Men e Cash only hanno una correlazione di 0.086807, che è una correlazione positiva molto debole. Questo suggerisce che non c’è quasi nessuna relazione tra gli uomini con comportamenti discriminatori e il divario salariale di genere considerando solo il reddito da lavoro in contanti.
- Any cash labor income e Cash only hanno una correlazione di 0.863380, che è una forte correlazione positiva. Questo indica che quando il divario salariale di genere considerando qualsiasi reddito da lavoro aumenta, anche il divario salariale di genere considerando solo il reddito da lavoro in contanti tende ad aumentare, e viceversa.)

### Cluster definition and visualization

Clustering is an unsupervised machine learning technique consisting in grouping unlabeled examples. The goal of this process is creating a data-driven classification of the observations and to infere relevant common features. The algorithm below deals with our datasets regarding gender gap across several fields. The first step is to create a syntetic visualization of the available data. <br>
Each dataframe will consider the most interesting statistics about these topics and will show them in an interesting way (da aggiustare)